3p/ampcontext.js

```diff
@@ -105,22 +112,25 @@ export class AmpContext {
   /**
    *  Parse the metadata attributes from the name and add them to
    *  the class instance.
+   *  @param {!Object|!String} contextData
    *  @private
    */
-  setupMetadata_() {
+  setupMetadata_(data) {
     try {
-      const data = JSON.parse(decodeURI(this.win_.name));
-      const context = data._context;
-      this.location = context.location;
-      this.canonicalUrl = context.canonicalUrl;
-      this.pageViewId = context.pageViewId;
-      this.sentinel = context.sentinel;
-      this.startTime = context.startTime;
-      this.referrer = context.referrer;
-    } catch (err) {
+      data = JSON.parse(decodeURI(data));
+    } catch (err) {}
+
+    if (!data || typeof data != 'object') {
       user().error('AMPCONTEXT', '- Could not parse metadata.');
```

Comment:
this is more likely a dev error? we actually can just skip it, since we are throwing in the next line.